### PR TITLE
Recommend Servlet

### PR DESCRIPTION
**Pull request by fang-helen**
Created a servlet to retrieve recommendations for a given user, or if none exist, compute them using only tags and distance for scoring. Also, add a doGet method to the survey servlet to check if user has completed the survey or not.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/RecommendServlet.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+          events.add(datastore.get(eventKey));
+        } catch (EntityNotFoundException eventNotFound) {
+          LOGGER.info("event " + eventId + " not found.");
+          limit++;
+        }
+      }
+    } catch (EntityNotFoundException | NullPointerException exception) {
+      // no prev recommendations, compute simple recommendations based on tags and distance only
+      LOGGER.info("No recommendations found for " + userID + ". Computing now.");
+      recommendations = new Entity("Recommendation", userID);
```
Comment (Yunni) [inline src/main/java/com/google/sps/servlets/RecommendServlet.java:82]: It's cleaner put the computation of recommendations into another method.

File: src/main/java/com/google/sps/servlets/RecommendServlet.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      Key userKey = KeyFactory.createKey("User", userID);
+      Entity userEntity = null;
+      try {
```
Comment (Yunni) [inline src/main/java/com/google/sps/servlets/RecommendServlet.java:85]: If possible, please avoid nested try-catch:
https://www.veracode.com/blog/secure-development/avoiding-nested-try-catch-java

File: src/main/java/com/google/sps/servlets/RecommendServlet.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+      } catch (EntityNotFoundException userNotFound) {
+        // user does not exist (no data)
+        userEntity = Utils.makeUserEntity(userID, true);
+      }
+    }
+    response.getWriter().println(gson.toJson(events));
+  }
+
+  /** Computes the correlation between a user and an event. */
+  private static double computeScore(
```
Comment (Yunni) [inline src/main/java/com/google/sps/servlets/RecommendServlet.java:127]: This method is hard to understand. Please add docstring to explain its behaviour.

File: src/main/java/com/google/sps/servlets/RecommendServlet.java (hunk 1 above)
Comment (fang-helen) [reply src/main/java/com/google/sps/servlets/RecommendServlet.java]: Fixed!

File: src/main/java/com/google/sps/servlets/RecommendServlet.java (hunk 2 above)
Comment (fang-helen) [reply src/main/java/com/google/sps/servlets/RecommendServlet.java:83]: Makes sense!

File: src/main/java/com/google/sps/servlets/RecommendServlet.java (hunk 3 above)
Comment (fang-helen) [reply src/main/java/com/google/sps/servlets/RecommendServlet.java:167]: Done!

**Pull request merged**